homeassistant/components/tis_control/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Config flow for TIS Control integration."""
+
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PORT
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TISConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TIS Control."""
+
+    async def async_step_user(self, user_input: dict | None = None) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            unique_id = f"tis_control:{user_input[CONF_PORT]}"
+            await self.async_set_unique_id(unique_id)
+            self._abort_if_unique_id_configured()
+
+            _LOGGER.info("Received user input %s", user_input)
+
+            # Validating the port user submitted
+            is_valid = self.validate_port(user_input[CONF_PORT])
+            if not is_valid:
+                errors["base"] = "invalid_port"  # Custom error key
+                _LOGGER.error("Provided port is invalid: %s", user_input[CONF_PORT])
+
+            if not errors:
+                return self.async_create_entry(
+                    title="TIS Control Bridge", data=user_input
+                )
+
+            # If there are errors, show the form again with the error message
+            _LOGGER.error("Errors occurred: %s", errors)
+            return self._show_setup_form(errors)
+
+        # If user_input is None (initial step), show the setup form
+        return self._show_setup_form(errors=errors)
+
+    @callback
+    def _show_setup_form(self, errors=None) -> ConfigFlowResult:
+        """Show the setup form to the user."""
+
+        schema = vol.Schema(
+            {vol.Required(CONF_PORT, default=6000): int},
```

Comment:
```suggestion
            {vol.Required(CONF_PORT, default=6000): cv.port},
```
https://github.com/home-assistant/core/blob/c9351a022ea28546afface55f2bc7d9d0d8de7d7/homeassistant/helpers/config_validation.py#L197

We have a validator for that.

Reply:
Thanks for your review, we have adjusted our config_flow and test_config_flow files to meet your suggested changes, feel free to review our changes and tell us if there is any other thing is required to be adjusted.